tests/test_recipe.py

```diff
@@ -0,0 +1,127 @@
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+"""
+Unit tests for sat.recipe module.
+"""
+
+from unittest.mock import Mock, PropertyMock, patch
+import unittest
+
+from sat.apiclient.vcs import VCSError
+from sat.recipe import (
+    HPCSoftwareRecipeCatalog,
+    HPCSoftwareRecipeError,
+    HPC_SOFTWARE_RECIPE_REPO_NAME,
+    HPC_SOFTWARE_RECIPE_REPO_ORG,
+    HPC_SOFTWARE_RECIPE_REPO_PATH
+)
+
+
+class TestHPCSoftwareRecipeCatalog(unittest.TestCase):
+    """Tests for the HPCSoftwareRecipeCatalog."""
+
+    def setUp(self):
+        self.mock_api_gw_host = 'api-gw-service-nmn.local'
+        self.mock_get_config_value = patch('sat.recipe.get_config_value').start()
+        self.mock_get_config_value.return_value = self.mock_api_gw_host
+        self.expected_clone_url = f'https://{self.mock_api_gw_host}{HPC_SOFTWARE_RECIPE_REPO_PATH}'
+
+        self.mock_vcs_repo_cls = patch('sat.recipe.VCSRepo').start()
+        self.mock_vcs_repo = self.mock_vcs_repo_cls.return_value
+        self.recipe_versions = ['22.03', '22.06', '22.06.1', '22.07.0', '22.07.1']
+        self.short_branch_names = [
+            # These branches follow the expected release branch format
+            f'{HPC_SOFTWARE_RECIPE_REPO_ORG}/{HPC_SOFTWARE_RECIPE_REPO_NAME}/{recipe_version}'
+            for recipe_version in self.recipe_versions
+        ] + [
+            # These branches do not follow the expect release branch format
+            'master',
+            'admin-created-branch'
+        ]
+        self.mock_refs = {f'refs/heads/{branch}': Mock() for branch in self.short_branch_names}
+        type(self.mock_vcs_repo).remote_refs = PropertyMock(return_value=self.mock_refs)
+
+    def tearDown(self):
+        patch.stopall()
+
+    def test_init(self):
+        """Test creation of a new HPCSoftwareRecipeCatalog."""
+        recipe_catalog = HPCSoftwareRecipeCatalog()
+        self.assertEqual(self.expected_clone_url, recipe_catalog.clone_url)
+        self.assertEqual(self.mock_vcs_repo, recipe_catalog.vcs_repo)
+
+    def test_recipes_property(self):
+        """Test the recipes property of HPCSoftwareRecipeCatalog"""
+        recipe_catalog = HPCSoftwareRecipeCatalog()
+
+        self.assertEqual(len(self.recipe_versions), len(recipe_catalog.recipes))
+        for recipe_version in self.recipe_versions:
+            self.assertIn(recipe_version, recipe_catalog.recipes)
+            recipe = recipe_catalog.recipes[recipe_version]
+            self.assertEqual(recipe_version, recipe.version)
+            self.assertEqual(f'cray/hpc-software-recipe/{recipe_version}',
+                             recipe.vcs_branch)
+            self.assertEqual(self.mock_vcs_repo, recipe.vcs_repo)
+
+    def test_recipes_property_vcs_error(self):
+        """Test the recipes property when there is an error accessing VCS."""
+        type(self.mock_vcs_repo).remote_refs = PropertyMock(side_effect=VCSError)
```

Comment:
Is `type(self.mock_vcs_repo)` just `MagicMock`? If so, could this affect other tests? I might be misunderstanding why modifying the class is needed in this instance.

Reply:
I'm not exactly sure how this works, but I believe the way I'm doing it is consistent with the [PropertyMock docs](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.PropertyMock).

Example of it not affecting other `MagicMock` objects:

```
>>> from unittest.mock import MagicMock, PropertyMock
>>> m1 = MagicMock()
>>> p = PropertyMock(return_value=3)
>>> type(m1).foo = p
>>> m1.foo
3
>>> m2 = MagicMock()
>>> m2.foo
<MagicMock name='mock.foo' id='4496337120'>
```